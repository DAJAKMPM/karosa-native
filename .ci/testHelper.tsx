import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
