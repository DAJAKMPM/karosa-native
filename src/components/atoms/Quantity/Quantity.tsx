/**
 *
 * Quantity
 * @format
 *
 */

import React, { FC, useState } from "react";

import type { PropsType } from "./types";
import QuantityStyles from "./styles";
import { View, TouchableOpacity } from "react-native";
import Text from "@app/atoms/Text";

const Quantity: FC<PropsType> = (props) => {
  const {} = props;

  const [quantity, setQuantity] = useState(0);

  const Increment = () => {
    setQuantity(quantity + 1);
  };

  const Decrement = () => {
    quantity <= 0 ? 0 : setQuantity(quantity - 1);
  };

  return (
    <View style={QuantityStyles.container}>
      <TouchableOpacity onPress={Decrement}>
        <View style={QuantityStyles.buttonDec}>
          <Text text={"-"} />
        </View>
      </TouchableOpacity>
      <View style={QuantityStyles.countStyle}>
        <Text text={quantity.toString()} />
      </View>
      <TouchableOpacity onPress={Increment}>
        <View style={QuantityStyles.buttonInc}>
          <Text text={"+"} />
        </View>
      </TouchableOpacity>
    </View>
  );
};

export default Quantity;
