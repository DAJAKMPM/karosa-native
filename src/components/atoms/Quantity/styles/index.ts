/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";

const QuantityStyles = StyleSheet.create({
  container: {
    justifyContent: "space-between",
    flexDirection: "row",
    borderColor: "grey",
    borderWidth: 0.5,
    width: 100,
  },
  buttonInc: {
    justifyContent: "center",
    flexDirection: "row",
    width: 24,
    borderLeftColor: "grey",
    borderLeftWidth: 0.5,
  },
  buttonDec: {
    justifyContent: "center",
    flexDirection: "row",
    width: 24,
    borderRightColor: "grey",
    borderRightWidth: 0.5,
  },
  countStyle: {
    alignContent: "center",
    paddingHorizontal: 10,
  },
});

export default QuantityStyles;
