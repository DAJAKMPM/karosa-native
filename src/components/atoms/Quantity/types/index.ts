/**
 *
 * @format
 *
 */

export type PropsType = {
  onDecrement(): void;
  onIncrement(): void;
  disableIncrement?: boolean;
  disableDecrement?: boolean;
  value: string;
};
