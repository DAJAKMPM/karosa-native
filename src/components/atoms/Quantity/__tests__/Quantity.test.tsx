import React from "react";
import { noop } from "lodash";
import { renderWithWrapper } from "@app/ci/testHelper";
import Quantity from "../Quantity";

describe("Quantity Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Quantity onDecrement={noop} onIncrement={noop} value="2" />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
