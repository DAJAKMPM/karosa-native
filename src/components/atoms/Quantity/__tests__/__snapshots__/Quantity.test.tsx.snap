// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 100,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#0AA351",
            "borderRadius": 3,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 18.75,
          }
        }
      >
        2
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#0AA351",
            "borderRadius": 3,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
